pilot/pkg/xds/ads.go

```diff
@@ -630,7 +630,7 @@ func setTopologyLabels(proxy *model.Proxy) {
 
 	locality := util.LocalityToString(proxy.Locality)
 	// add topology labels to proxy labels
-	proxy.Labels = labelutil.AugmentLabels(proxy.Labels, proxy.Metadata.ClusterID, locality, proxy.Metadata.Network)
+	proxy.Labels = labelutil.AugmentLabels(proxy.Labels, proxy.Metadata.ClusterID, locality, "", proxy.Metadata.Network)
```

Comment:
In addition to modifying Labels we should have a `proxy.NodeName` we set, just like L624